.gitlab-ci.yml

```diff
@@ -6,30 +6,169 @@ image:
   name: mcr.microsoft.com/dotnet/sdk:6.0
   entrypoint: [""]
 
-cache:
-  key: "$CI_JOB_STAGE-$CI_COMMIT_REF_SLUG"
-  paths:
-    - "./packages/pangea-sdk/PangeaCyber.Net/obj/project.assets.json"
-    - "./packages/pangea-sdk/PangeaCyber.Net/obj/*.csproj.nuget.*"
-    - ".nuget"
-  policy: pull-push
-
-before_script:
-  - cd packages/pangea-sdk
-  - dotnet restore --packages .nuget
-  - dotnet build --no-restore --configuration Release
-
-lint:
+.dotnet-sdk-base:
+  cache:
+    key: "$CI_JOB_STAGE-$CI_COMMIT_REF_SLUG"
+    paths:
+      - "./packages/pangea-sdk/PangeaCyber.Net/obj/project.assets.json"
+      - "./packages/pangea-sdk/PangeaCyber.Net/obj/*.csproj.nuget.*"
+      - ".nuget"
+    policy: pull-push
+
+  before_script:
+    - cd packages/pangea-sdk
+    - dotnet restore --packages .nuget
+    - dotnet build --no-restore --configuration Release
+
+.dotnet-sdk-test-base:
+  stage: integration_tests
+  cache:
+    key: "$CI_JOB_STAGE-$CI_COMMIT_REF_SLUG"
+    paths:
+      - "./packages/pangea-sdk/PangeaCyber.Net/obj/project.assets.json"
+      - "./packages/pangea-sdk/PangeaCyber.Net/obj/*.csproj.nuget.*"
+      - ".nuget"
+    policy: pull-push
+
+  before_script:
+    - echo $ENV
+    - echo $CLOUD
+  # Update environment variables
+  # Domain
+    - export PANGEA_INTEGRATION_DOMAIN_${ENV}="$(eval echo \$PANGEA_INTEGRATION_DOMAIN_${ENV}_${CLOUD})"
+  # Tokens
+    - export PANGEA_INTEGRATION_TOKEN_${ENV}="$(eval echo \$PANGEA_INTEGRATION_TOKEN_${ENV}_${CLOUD})"
+    - export PANGEA_INTEGRATION_VAULT_TOKEN_${ENV}="$(eval echo \$PANGEA_INTEGRATION_VAULT_TOKEN_${ENV}_${CLOUD})"
+    - export PANGEA_INTEGRATION_CUSTOM_SCHEMA_TOKEN_${ENV}="$(eval echo \$PANGEA_INTEGRATION_CUSTOM_SCHEMA_TOKEN_${ENV}_${CLOUD})"
+    - export PANGEA_INTEGRATION_MULTI_CONFIG_TOKEN_${ENV}="$(eval echo \$PANGEA_INTEGRATION_MULTI_CONFIG_TOKEN_${ENV}_${CLOUD})"
+  # Config IDs
+    - export PANGEA_AUDIT_CONFIG_ID_1_${ENV}="$(eval echo \$PANGEA_AUDIT_CONFIG_ID_1_${ENV}_${CLOUD})"
+    - export PANGEA_AUDIT_CONFIG_ID_2_${ENV}="$(eval echo \$PANGEA_AUDIT_CONFIG_ID_2_${ENV}_${CLOUD})"
+    - export PANGEA_REDACT_CONFIG_ID_1_${ENV}="$(eval echo \$PANGEA_REDACT_CONFIG_ID_1_${ENV}_${CLOUD})"
+    - export PANGEA_REDACT_CONFIG_ID_2_${ENV}="$(eval echo \$PANGEA_REDACT_CONFIG_ID_2_${ENV}_${CLOUD})"
+
+    - cd packages/pangea-sdk
+    - dotnet restore --packages .nuget
+    - dotnet build --no-restore --configuration Release
+
+  parallel:
+    matrix:
+      - CLOUD: [AWS, GCP]
+
+  rules:
+    - if: $CI_PIPELINE_SOURCE == "push"
+
+dotnet-sdk-lint:
+  extends: .dotnet-sdk-base
   stage: lint
   script:
     - dotnet format --verify-no-changes
 
-tests:
-  stage: test
+dotnet-sdk-it-audit:
+  extends: .dotnet-sdk-test-base
+  variables:
+    ENV: ${SERVICE_AUDIT_ENV}
+  script:
```

Comment:
I know @kenany had a setup in gitlab that ended up dynamically making this test matrix. Not sure if there's an easy way to do that on the GitLab side.

Reply:
I've been thinking about that but here there is a difference here that we should set this `ENV` variable here each time, so it's not the same case but I guess in the future we'll figure it how to solve it and simplify it